modules/state/s3.tf

```diff
@@ -64,3 +64,35 @@ resource "aws_s3_bucket_public_access_block" "public_access_block_jenkins" {
   restrict_public_buckets = true
 }
 
+resource "aws_s3_bucket" "tdr_terraform_state_grafana" {
+  bucket = "tdr-terraform-state-grafana"
+  acl    = "private"
+
+  server_side_encryption_configuration {
+    rule {
+      apply_server_side_encryption_by_default {
+        sse_algorithm = "AES256"
+      }
+    }
+  }
+
+  versioning {
+    enabled = true
+  }
+
+  tags = merge(
+    var.common_tags,
+    map(
+      "Name", "TDR Grafana Terraform State",
+    )
+  )
+}
+
+resource "aws_s3_bucket_public_access_block" "public_access_block_grafana" {
```

Comment:
The S3 bucket and public access blocks are very similar for Grafana, Jenkins and... whatever the first one is. (TDR environments?) Is it worth commonising them? Is it possible? I don't know if Terraform would force you to recreate the S3 bucket or if it could keep the old one.

Reply:
I noticed this as well but was a bit hesitant about changing it, for fear of breaking something. Will take a look though.

Reply:
This is actually a bit more complicated than it looks, as the policy is tied to a specific S3 bucket id which is one of the params that needs to be included in the resources. I'm a bit reluctant to get into this at the moment.